call-notes/call_5.md

```diff
@@ -0,0 +1,152 @@
+-|-
+
+**Maximilian Schneider:** Yeah, so for the QUIC tuning parameters, that was like my request which was that we have someone who worked on it previously lead the discussion. I don't know if someone is here that worked on it. And I can only give the outside perspective of reverse engineering some of those things. But you know, reverse engineering requirements is very difficult sometimes and that was the main request that we get someone who knows all the requirements, maybe to present them. But if there is no one here, then maybe we put it for next time.
+
+**Jacob Creech:** Stephen Akridge, could you talk a little bit how you all choose tuning for QUIC.
+
+**Stephen Akridge:** Yeah, I can talk a little bit. Yeah, we had a somewhat specific performance idea in mind right. We want it to be in that like 50k to 100K TPS range in the frontend or at most try to split the streams that way. And allow right enough streams for each client to be able to fit within that. So, you know it was mainly from like benchmarking in the test environment and seeing kind of what bandwidth we could obtain unlike a somewhat optimal connection right which would be like the worst case in terms of TPS in like the real validator. And trying to calibrate that to what we thought like the backend behind QUIC could sustain or handle. And also like keeping the memory within something that was feasible for like a normal validator. So obviously, you have got more streams, you need to have more memory to hold all the packets that are kind of in flight. And potentially reconstruct a transaction that might cross multiple packets. So the more outstanding streams you allow, the more memory you need to frame all those incoming streams.
+
+**Maximilian Schneider:** Yeah, one limit we found and I think we were curious about was I think it's like 2,000 transactions every 50 milliseconds like a leaky bucket model.
+
+That would mean that's just to verify concentrate.
+
+**Stephen Akridge:** Yeah, I think like we were talking about in Discord right that's not like a hard limit. The 50 milliseconds is kind of a target for that stage to not like that's the time it is spending to verify and things are coming in behind it right into the channel to get backed up. So we want to make sure that we are pinging the channel like pretty frequently like within 100 milliseconds to make sure that the channel can back up faster than the incoming packet flow. So, I mean it doesn't have a hard requirement right like you have a machine that can basically verify 10,000 transactions in 50 milliseconds. It will clear all of those and it will bring up the new batch and it will verify it. But it is just a way to keep the queue from doing an exponential fill-up. If you have a large batch and you let it fill in behind you and that is even larger than the next batch then you are just like an unbounded condition and that it's not good for the memory in the validator.
```

Comment:
```suggestion
**Stephen Akridge:** That's just to verify constants right. Yeah, I think like we were talking about in Discord right that's not like a hard limit. The 50 milliseconds is kind of a target for that stage to not like that's the time it is spending to verify and things are coming in behind it right into the channel to get backed up. So we want to make sure that we are pinging the channel like pretty frequently like within 100 milliseconds to make sure that the channel can back up faster than the incoming packet flow. So, I mean it doesn't have a hard requirement right like you have a machine that can basically verify 10,000 transactions in 50 milliseconds. It will clear all of those and it will bring up the new batch and it will verify it. But it is just a way to keep the queue from doing an exponential fill-up. If you have a large batch and you let it fill in behind you and that is even larger than the next batch then you are just like an unbounded condition and that it's not good for the memory in the validator.
```